authentication/basicAuth.js

```
@@ -0,0 +1,68 @@
+
+const { User } = require("../models/userModel.js");
+function authUsered(req, res, next) {
+  if (req.body == null) {
+    res.status(401);
+    return res.send("you need to sign in");
+  }
+
+  next();
+}
+
+function adminRole() {
+  return async (req, res, next) => {
+    const { email } = req.body;
+    const user = await User.findOne({ email });
+    if (!user) {
+      res.send("Oops,we dont have you in our registery plz regsiter first ");
```

Don't use short form like plz. Also please throw an error and handle it using try catch